console/console-init/ui/src/modules/device/components/DeviceFilter/DeviceFilter.tsx

```diff
@@ -123,21 +128,25 @@ const DeviceFilter: React.FunctionComponent<IDeviceFilterProps> = (
       key: "disabled",
       value: "disabled",
       label: "Disabled"
+    },
+    {
+      key: "allStatus",
+      value: "allStatus",
+      label: "All status"
     }
   ];
   const onClearFilter = () => {
-    setFilter(setInitialFilter());
+    setFilter(getInitialFilter());
     setIsKebabOpen(false);
   };
   const onRedoFilter = () => {
     // redoFilter();
     const length = lastAppliedFilter.length;
-    const data = JSON.parse(
-      JSON.stringify({ ...lastAppliedFilter[length - 2] })
-    );
+    const data = createDeepCopy({ ...lastAppliedFilter[length - 2] });
```

Comment:
Please rename data and datalist to be more descriptive.

Reply:
done